eslint.config.mjs

```diff
@@ -0,0 +1,394 @@
+              position: 'after',
+            },
+            {
+              pattern:
+                '{classnames,react-helmet,react-router,react-router-dom}',
+              group: 'external',
+              position: 'before',
+            },
+            {
+              pattern:
+                '{immutable,@reduxjs/toolkit,react-redux,react-immutable-proptypes,react-immutable-pure-component}',
+              group: 'external',
+              position: 'before',
+            },
+            {
+              pattern: '{mastodon/**}',
+              group: 'internal',
+              position: 'after',
+            },
+          ],
+
+          pathGroupsExcludedImportTypes: [],
+        },
+      ],
+
+      'promise/always-return': 'off',
+      'promise/catch-or-return': [
+        'error',
+        {
+          allowFinally: true,
+        },
+      ],
+      'promise/no-callback-in-promise': 'off',
+      'promise/no-nesting': 'off',
+      'promise/no-promise-in-callback': 'off',
+
+      'react/jsx-filename-extension': [
+        'error',
+        {
+          extensions: ['.jsx', 'tsx'],
+        },
+      ],
+
+      'react/jsx-boolean-value': 'error',
+      'react/display-name': 'off',
+      'react/jsx-fragments': ['error', 'syntax'],
+      'react/jsx-equals-spacing': 'error',
+      'react/jsx-no-bind': 'error',
+      'react/jsx-no-useless-fragment': 'error',
+      'react/jsx-no-target-blank': [
+        'error',
+        {
+          allowReferrer: true,
+        },
+      ],
+      'react/jsx-tag-spacing': 'error',
+      'react/jsx-wrap-multilines': 'error',
+      'react/self-closing-comp': 'error',
+    },
+  },
+  {
+    files: [
+      'app/javascript/mastodon/common.js',
+      'app/javascript/mastodon/features/emoji/unicode_to_unified_name.js',
+      'app/javascript/mastodon/features/emoji/emoji_compressed.js',
+      'app/javascript/mastodon/features/emoji/unicode_to_filename.js',
+      'app/javascript/mastodon/service_worker/web_push_locales.js',
+      '**/*.config.js',
+      '**/.*rc.js',
+      '**/ide-helper.js',
+      'config/webpack/**/*',
+      'config/formatjs-formatter.js',
+    ],
+
+    languageOptions: {
+      globals: {
+        ...globals.commonjs,
+        ...globals.node,
+      },
+
+      ecmaVersion: 5,
+      sourceType: 'commonjs',
+    },
+
+    rules: {
+      'import/no-commonjs': 'off',
+    },
+  },
+  {
+    files: ['**/*.ts', '**/*.tsx'],
+
+    extends: [
+      js.configs.recommended,
+      tseslint.configs.strictTypeChecked,
+      tseslint.configs.stylisticTypeChecked,
+      react.configs.flat.recommended,
+      react.configs.flat['jsx-runtime'],
+      reactHooks.configs['recommended-latest'],
+      jsxA11Y.flatConfigs.recommended,
+      importPlugin.flatConfigs.recommended,
+      importPlugin.flatConfigs.react,
+      importPlugin.flatConfigs.typescript,
+      promisePlugin.configs['flat/recommended'],
+      jsdoc.configs['flat/recommended-typescript'],
```

Comment:
We probably dont need to extends from all of those here, only the one that are not set as top-level configs.

Reply:
I think you're probably right, but the thing holding me back from just removing it is the "default" files that some of these extensions aren't consistent. EX: Some set `file` globs, and others don't, and may just target JS/MJS files. The config inspector might be able to unravel that though